src/AsyncTCP.cpp

```diff
@@ -65,7 +65,7 @@ extern "C" {
 
 // https://github.com/espressif/arduino-esp32/issues/10526
 namespace {
-#ifdef CONFIG_LWIP_TCPIP_CORE_LOCKING
+#if defined CONFIG_LWIP_TCPIP_CORE_LOCKING && CONFIG_LWIP_TCPIP_CORE_LOCKING
```

Comment:
Is it to avoid using that if -DCONFIG_LWIP_TCPIP_CORE_LOCKING=0 is set ?
If so, should it be instead more clear and be: `#if CONFIG_LWIP_TCPIP_CORE_LOCKING  == 1` ?

Reply:
> I do not have Libretiny's at hand,

neither I...

So maybe it is better to leave it like you did to handle all the combinations...

Or we do like arduino: `ifdef CONFIG_LWIP_TCPIP_CORE_LOCKING`, but this opens the door to unexpected behavior like you mentioned with `CONFIG_LWIP_TCPIP_CORE_LOCKING=0`

Probably better to stick with what you did...